.. _backbone-label:

Nextpolish
==============================

Introduction
~~~~~~~~
NextPolish is used to fix base errors (SNV/Indel) in the genome generated by noisy long reads, it can be used with short read data only or long read data only or a combination of both. It contains two core modules, and use a stepwise fashion to correct the error bases in reference genome. To correct/assemble the raw third-generation sequencing (TGS) long reads with approximately 10-15% sequencing errors, please use NextDenovo.


| For more information, please check:
| Home page: https://github.com/Nextomics/NextPolish

Versions
.. list-table::
  :width: 100 %
  :widths: 25 75
  :header-rows: 1

  * - Cluster
    - Versions

  * - ANVIL
    - 1.4.1
  * - BELL
    - 1.4.1
  * - GAUTSCHI
    - 1.4.1
  * - NEGISHI
    - 1.4.1
  * - SCHOLAR
    - 1.4.1

Commands
~~~~~~~
- nextPolish

Module
~~~~~~~~
You can load the modules by::

    module load biocontainers
    module load nextpolish

Example job
~~~~~
.. warning::
    Using ``#!/bin/sh -l`` as shebang in the slurm job script will cause the failure of some biocontainer modules. Please use ``#!/bin/bash`` instead.

To run nextpolish on our clusters::

    #!/bin/bash
    #SBATCH -A myallocation     # Allocation name
    #SBATCH -t 1:00:00
    #SBATCH -N 1
    #SBATCH -n 1
    #SBATCH --job-name=nextpolish
    #SBATCH --mail-type=FAIL,BEGIN,END
    #SBATCH --error=%x-%J-%u.err
    #SBATCH --output=%x-%J-%u.out

    module --force purge
    ml biocontainers nextpolish
