.. _backbone-label:  

UCSC Executables
============================== 

Introduction
~~~~~~~
``UCSC Executables`` is a variety of executables that perform functions ranging from sequence analysis and format conversion, to basic number crunching and statistics, to complex database generation and manipulation.  

These executables have been downloaded from http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64.v369/ and made available on RCAC clusters.  

Versions
~~~~~~~
.. list-table::
  :width: 100 %
  :widths: 25 75
  :header-rows: 1

  * - Cluster
    - Version(s)

  * - ANVIL
    - 369
  * - BELL
    - 369
  * - GAUTSCHI
    - 369
  * - NEGISHI
    - 369
  * - SCHOLAR
    - 369

Commands
~~~~~~
- addCols
- ameme
- autoDtd
- autoSql
- autoXml
- ave
- aveCols
- axtChain
- axtSort
- axtSwap
- axtToMaf
- axtToPsl
- bamToPsl
- barChartMaxLimit
- bedClip
- bedCommonRegions
- bedCoverage
- bedExtendRanges
- bedGeneParts
- bedGraphPack
- bedGraphToBigWig
- bedIntersect
- bedItemOverlapCount
- bedJoinTabOffset
- bedJoinTabOffset.py
- bedMergeAdjacent
- bedPartition
- bedPileUps
- bedRemoveOverlap
- bedRestrictToPositions
- bedSingleCover.pl
- bedSort
- bedToBigBed
- bedToExons
- bedToGenePred
- bedToPsl
- bedWeedOverlapping
- bigBedInfo
- bigBedNamedItems
- bigBedSummary
- bigBedToBed
- bigGenePredToGenePred
- bigHeat
- bigMafToMaf
- bigPslToPsl
- bigWigAverageOverBed
- bigWigCat
- bigWigCluster
- bigWigCorrelate
- bigWigInfo
- bigWigMerge
- bigWigSummary
- bigWigToBedGraph
- bigWigToWig
- binFromRange
- blastToPsl
- blastXmlToPsl
- blat
- calc
- catDir
- catUncomment
- chainAntiRepeat
- chainBridge
- chainCleaner
- chainFilter
- chainMergeSort
- chainNet
- chainPreNet
- chainScore
- chainSort
- chainSplit
- chainStitchId
- chainSwap
- chainToAxt
- chainToPsl
- chainToPslBasic
- checkAgpAndFa
- checkCoverageGaps
- checkHgFindSpec
- checkTableCoords
- chopFaLines
- chromGraphFromBin
- chromGraphToBin
- chromToUcsc
- clusterGenes
- clusterMatrixToBarChartBed
- colTransform
- countChars
- cpg_lh
- crTreeIndexBed
- crTreeSearchBed
- dbSnoop
- dbTrash
- endsInLf
- estOrient
- expMatrixToBarchartBed
- faAlign
- faCmp
- faCount
- faFilter
- faFilterN
- faFrag
- faNoise
- faOneRecord
- faPolyASizes
- faRandomize
- faRc
- faSize
- faSomeRecords
- faSplit
- faToFastq
- faToTab
- faToTwoBit
- faToVcf
- faTrans
- fastqStatsAndSubsample
- fastqToFa
- featureBits
- fetchChromSizes
- findMotif
- fixStepToBedGraph.pl
- gapToLift
- genePredCheck
- genePredFilter
- genePredHisto
- genePredSingleCover
- genePredToBed
- genePredToBigGenePred
- genePredToFakePsl
- genePredToGtf
- genePredToMafFrames
- genePredToProt
- gensub2
- getRna
- getRnaPred
- gff3ToGenePred
- gff3ToPsl
- gmtime
- gtfToGenePred
- headRest
- hgBbiDbLink
- hgFakeAgp
- hgFindSpec
- hgGcPercent
- hgGoldGapGl
- hgLoadBed
- hgLoadChain
- hgLoadGap
- hgLoadMaf
- hgLoadMafSummary
- hgLoadNet
- hgLoadOut
- hgLoadOutJoined
- hgLoadSqlTab
- hgLoadWiggle
- hgSpeciesRna
- hgTrackDb
- hgWiggle
- hgsql
- hgsqldump
- hgvsToVcf
- hicInfo
- htmlCheck
- hubCheck
- hubClone
- hubPublicCheck
- ixIxx
- lastz-1.04.00
- lastz_D-1.04.00
- lavToAxt
- lavToPsl
- ldHgGene
- liftOver
- liftOverMerge
- liftUp
- linesToRa
- localtime
- mafAddIRows
- mafAddQRows
- mafCoverage
- mafFetch
- mafFilter
- mafFrag
- mafFrags
- mafGene
- mafMeFirst
- mafNoAlign
- mafOrder
- mafRanges
- mafSpeciesList
- mafSpeciesSubset
- mafSplit
- mafSplitPos
- mafToAxt
- mafToBigMaf
- mafToPsl
- mafToSnpBed
- mafsInRegion
- makeTableList
- maskOutFa
- matrixClusterColumns
- matrixMarketToTsv
- matrixNormalize
- mktime
- mrnaToGene
- netChainSubset
- netClass
- netFilter
- netSplit
- netSyntenic
- netToAxt
- netToBed
- newProg
- newPythonProg
- nibFrag
- nibSize
- oligoMatch
- overlapSelect
- para
- paraFetch
- paraHub
- paraHubStop
- paraNode
- paraNodeStart
- paraNodeStatus
- paraNodeStop
- paraSync
- paraTestJob
- parasol
- positionalTblCheck
- pslCDnaFilter
- pslCat
- pslCheck
- pslDropOverlap
- pslFilter
- pslHisto
- pslLiftSubrangeBlat
- pslMap
- pslMapPostChain
- pslMrnaCover
- pslPairs
- pslPartition
- pslPosTarget
- pslPretty
- pslRc
- pslRecalcMatch
- pslRemoveFrameShifts
- pslReps
- pslScore
- pslSelect
- pslSomeRecords
- pslSort
- pslSortAcc
- pslStats
- pslSwap
- pslToBed
- pslToBigPsl
- pslToChain
- pslToPslx
- pslxToFa
- qaToQac
- qacAgpLift
- qacToQa
- qacToWig
- raSqlQuery
- raToLines
- raToTab
- randomLines
- rmFaDups
- rowsToCols
- sizeof
- spacedToTab
- splitFile
- splitFileByColumn
- sqlToXml
- strexCalc
- stringify
- subChar
- subColumn
- tabQuery
- tailLines
- tdbQuery
- tdbRename
- tdbSort
- textHistogram
- tickToDate
- toLower
- toUpper
- trackDbIndexBb
- transMapPslToGenePred
- trfBig
- twoBitDup
- twoBitInfo
- twoBitMask
- twoBitToFa
- ucscApiClient
- udr
- vai.pl
- validateFiles
- validateManifest
- varStepToBedGraph.pl
- webSync
- wigCorrelate
- wigEncode
- wigToBigWig
- wordLine
- xmlCat
- xmlToSql

Module
~~~~~~~
You can load the modules by::

    module load biocontainers
    module load ucsc_genome_toolkit/369

Example job
~~~~~~
.. warning::
    Using ``#!/bin/sh -l`` as shebang in the slurm job script will cause the failure of some biocontainer modules. Please use ``#!/bin/bash`` instead.

To run UCSC executables on our our clusters::

    #!/bin/bash
    #SBATCH -A myallocation     # Allocation name 
    #SBATCH -t 1:00:00
    #SBATCH -N 1
    #SBATCH -n 12
    #SBATCH --job-name=UCSC
    #SBATCH --mail-type=FAIL,BEGIN,END
    #SBATCH --error=%x-%J-%u.err
    #SBATCH --output=%x-%J-%u.out

    module --force purge
    ml biocontainers ucsc_genome_toolkit/369
    
    blat genome.fasta input.fasta blat.out
    fastqToFa input.fastq  output.fasta 
