.. _backbone-label:

Transdecoder
==============================

Introduction
~~~~~~~~
Transdecoder identifies candidate coding regions within transcript sequences, such as those generated by de novo RNA-Seq transcript assembly using Trinity, or constructed based on RNA-Seq alignments to the genome using Tophat and Cufflinks.


| For more information, please check:
| BioContainers: https://biocontainers.pro/tools/transdecoder 
| Home page: https://github.com/TransDecoder/TransDecoder

.. note::
    Please follow the recommended citation guidelines from the developers when you use the tool in research.

Versions
~~~~~~~
.. list-table::
  :width: 100 %
  :widths: 25 75
  :header-rows: 1

  * - Cluster
    - Version(s)

  * - ANVIL
    - 5.5.0
  * - BELL
    - 5.5.0
  * - GAUTSCHI
    - 5.5.0
  * - NEGISHI
    - 5.5.0
  * - SCHOLAR
    - 5.5.0

Commands
~~~~~~~
- TransDecoder.LongOrfs
- TransDecoder.Predict
- cdna_alignment_orf_to_genome_orf.pl
- compute_base_probs.pl
- exclude_similar_proteins.pl
- fasta_prot_checker.pl
- ffindex_resume.pl
- gene_list_to_gff.pl
- get_FL_accs.pl
- get_longest_ORF_per_transcript.pl
- get_top_longest_fasta_entries.pl
- gff3_file_to_bed.pl
- gff3_file_to_proteins.pl
- gff3_gene_to_gtf_format.pl
- gtf_genome_to_cdna_fasta.pl
- gtf_to_alignment_gff3.pl
- gtf_to_bed.pl
- nr_ORFs_gff3.pl
- pfam_runner.pl
- refine_gff3_group_iso_strip_utrs.pl
- refine_hexamer_scores.pl
- remove_eclipsed_ORFs.pl
- score_CDS_likelihood_all_6_frames.pl
- select_best_ORFs_per_transcript.pl
- seq_n_baseprobs_to_loglikelihood_vals.pl
- start_codon_refinement.pl
- train_start_PWM.pl
- uri_unescape.pl

Module
~~~~~~~~
You can load the modules by::

    module load biocontainers
    module load transdecoder

Example job
~~~~~
.. warning::
    Using ``#!/bin/sh -l`` as shebang in the slurm job script will cause the failure of some biocontainer modules. Please use ``#!/bin/bash`` instead.

To run transdecoder on our clusters::

.. tab-set::

   .. tab-item:: Anvil

      .. code-block:: bash

        #!/bin/bash
        #SBATCH -A myallocation     # Allocation name
        #SBATCH -p wholenode        # Partition name
        #SBATCH -t 1:00:00
        #SBATCH -N 1
        #SBATCH -n 1
        #SBATCH --job-name=transdecoder
        #SBATCH --mail-type=FAIL,BEGIN,END
        #SBATCH --error=%x-%J-%u.err
        #SBATCH --output=%x-%J-%u.out

        module --force purge
        module biocontainers transdecoder

   .. tab-item:: Bell/Gautschi/Negishi

      .. code-block:: bash

        #!/bin/bash
        #SBATCH -A mygroup     # Group name
        #SBATCH -p cpu         # Partition name
        #SBATCH -q normal      # QOS name (optional)
        #SBATCH -t 1:00:00
        #SBATCH -N 1
        #SBATCH -n 1
        #SBATCH --job-name=transdecoder
        #SBATCH --mail-type=FAIL,BEGIN,END
        #SBATCH --error=%x-%J-%u.err
        #SBATCH --output=%x-%J-%u.out

        module --force purge
        module biocontainers transdecoder

   .. tab-item:: Gautschi-AI/Gilbreth

      .. code-block:: bash

        #!/bin/bash
        #SBATCH -A mygroup     # Group name
        #SBATCH -p a100        # Partition name
        #SBATCH --gres=gpu:1   # Number of GPUs
        #SBATCH -q normal      # QOS name (optional)
        #SBATCH -t 1:00:00
        #SBATCH -N 1
        #SBATCH -n 1
        #SBATCH --job-name=transdecoder
        #SBATCH --mail-type=FAIL,BEGIN,END
        #SBATCH --error=%x-%J-%u.err
        #SBATCH --output=%x-%J-%u.out

        module --force purge
        module biocontainers transdecoder

   .. tab-item:: Scholar

      .. code-block:: bash

        #!/bin/bash
        #SBATCH -A queue     # Queue name
        #SBATCH -t 1:00:00
        #SBATCH -N 1
        #SBATCH -n 1
        #SBATCH --job-name=transdecoder
        #SBATCH --mail-type=FAIL,BEGIN,END
        #SBATCH --error=%x-%J-%u.err
        #SBATCH --output=%x-%J-%u.out

        module --force purge
        module biocontainers transdecoder
